src/NuGet.Clients/VisualStudio.Facade/Events/IRestoreEvents.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.VisualStudio.Facade.Events
+{
+    public delegate void SolutionRestoreCompletedEventHandler(SolutionRestoredEventArgs args);
+
+    /// <summary>
+    /// The interface for listening to events concerning restore operations.
+    /// </summary>
+    public interface IRestoreEvents
+    {
```

Comment:
At @alpaix suggestion. I found the smallest common project to put this in. If it's in NuGet.Tools, then the NuGetConsole.Host.PowerShell and NuGet.SolutionRestoreManager projects will not see it (circular dependency).